File: examples/android/CMeSdkJava/app/src/main/java/me/connect/sdk/java/sample/history/HistoryViewModel.java
package me.connect.sdk.java.sample.history;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

import me.connect.sdk.java.sample.db.Database;
import me.connect.sdk.java.sample.db.entity.Action;

import static me.connect.sdk.java.sample.db.ActionStatus.HISTORIZED;

public class HistoryViewModel extends AndroidViewModel {
    private final Database db;
    private LiveData<List<Action>> actions;

    public HistoryViewModel(@NonNull Application application) {
        super(application);
        db = Database.getInstance(application);
    }

    public LiveData<List<Action>> getHistory() {
        if (actions == null) {
            actions = db.actionDao().getActionsByStatus(HISTORIZED.toString());
        }
        return actions;
    }
}

File: examples/android/CMeSdkJava/app_kt/src/main/kotlin/me/connect/sdk/java/samplekt/history/HistoryViewModel.kt
package me.connect.sdk.java.samplekt.history

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import me.connect.sdk.java.samplekt.db.Database
import me.connect.sdk.java.samplekt.db.entity.Action

import me.connect.sdk.java.samplekt.db.ActionStatus.HISTORIZED

class HistoryViewModel(application: Application) : AndroidViewModel(application) {
    private var db = Database.getInstance(application)
    private val actionsLiveData: LiveData<List<Action>> by lazy {
        db.actionDao().getActionsByStatus(HISTORIZED.toString())
    }

    fun getHistory(): LiveData<List<Action>> = actionsLiveData
}